src/functional/geometry/convex_hull.zig

```diff
@@ -66,22 +72,127 @@ pub const ConvexHull = struct {
 
     /// Returns the points in the `upper_hull` except the last one.
     pub fn getUpperHullExceptLast(self: *ConvexHull) []const DiscretePoint {
-        const len = self.upper_hull.items.len;
-        if (len <= 1) {
+        const upper_hull_len = self.upper_hull.items.len;
+        if (upper_hull_len <= 1) {
             // Return an empty array if there's only one or no items.
             return &[_]DiscretePoint{};
         }
-        return self.upper_hull.items[0 .. len - 1];
+        return self.upper_hull.items[0 .. upper_hull_len - 1];
     }
 
     /// Returns the points in the `lower_hull` except the last one.
     pub fn getLowerHullExceptLast(self: *ConvexHull) []const DiscretePoint {
-        const len = self.lower_hull.items.len;
-        if (len <= 1) {
+        const lower_hull_len = self.lower_hull.items.len;
+        if (lower_hull_len <= 1) {
             // Return an empty array if there's only one or no items.
             return &[_]DiscretePoint{};
         }
-        return self.lower_hull.items[0 .. len - 1];
+        return self.lower_hull.items[0 .. lower_hull_len - 1];
+    }
+
+    /// Computes the `LinearFunction` with minimum square error based on the Minimum-Area Bounding
+    /// Rectangle (MABR) of a Convex Hull. The MABR is computed using the Rotating Calipers
+    /// algorithm described in the paper:
+    /// Shamos, Michael (1978). "Computational Geometry" (PDF). Yale University. pp. 76–81.
+    /// Our implementation is based mostly on the description found in:
+    /// https://en.wikipedia.org/wiki/Rotating_calipers.
+    pub fn computeMABRLinearFunction(self: *ConvexHull) Error!LinearFunction {
+        const hullPoints: ArrayList(DiscretePoint) = try self.getAllPoints();
+        defer hullPoints.deinit();
+        const convex_hull_len: usize = hullPoints.items.len;
+
+        // One point in the ConvexHull, the line has `slope=0`.
```

Comment:
Unclear what slope equals zero means in this context, is it simply because there is only one point for the function? Maybe expand their explanation.

Reply:
fixed.